### PR TITLE
checkpoint and restore from checkpoint

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yu745514916@live.com>

### CODE REVIEW

File: tiup/pkg/cluster/manager/pCloud.go [hunk 1]
```diff
@@ -213,13 +327,13 @@ func (m *Manager) Backup2Cloud(name string, opt operator.Options) error {
 		if err != nil {
 			return err
 		}
-		fmt.Println(color.GreenString("Starting streaming.."))
-		err = m.StartsIncrementalBackup(pdHost, metadata, clusterID)
+		fmt.Println(color.GreenString("Starting upload.."))
+		err = m.DoBackup(info, clusterID)
```
Comment (3pointer) [inline tiup/pkg/cluster/manager/pCloud.go]: `StartsIncrementalBackup` happen before `DoBackup` can avoid data loss

File: tiup/pkg/cluster/manager/pCloud.go [hunk 2]
```diff
@@ -252,25 +366,37 @@ func (m *Manager) SaveToFile(file string, content string) error {
...
-		}
-	})
-	if len(pdHost) == 0 {
-		return errors.New("cluster doesn't have any cdc server")
+	info := m.getClusterInfo(name)
+	if err := info.AssertPDExists(); err != nil {
+		return err
+	}
+	clusterFile := path.Join(cloudDir, authKeyForCluster(name), "cloudFile")
+	clusterID, err := os.ReadFile(clusterFile)
```
Comment (3pointer) [inline tiup/pkg/cluster/manager/pCloud.go]: what if clusterFile not exists?🤔

File: tiup/pkg/cluster/manager/pCloud.go (hunk 1 above)
Comment (YuJuncen) [reply tiup/pkg/cluster/manager/pCloud.go]: Yep...🤔

File: tiup/pkg/cluster/manager/pCloud.go (hunk 2 above)
Comment (YuJuncen) [reply tiup/pkg/cluster/manager/pCloud.go]: I guess it would fail and stop?

**Pull request merged**